lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs

```diff
@@ -387,6 +395,50 @@ testRule({
 				'the repeat() notation has non-trivial semantics and is currently not fixable (rows)',
 			message: messages.expected('grid-template'),
 		},
+		{
+			code: 'a { text-decoration-color: purple; text-decoration-thickness: 1px; text-decoration-style: solid; text-decoration-line: underline; }',
+			fixed: 'a { text-decoration: underline solid purple 1px; }',
+			message: messages.expected('text-decoration'),
+			description: 'CSS Text Decoration Module Level 4',
+		},
+	],
+});
+
+testRule({
+	ruleName,
+	config: [
+		true,
+		{
+			ignoreLonghands: [
+				'background-size',
+				'background-origin',
+				'background-clip',
+				'text-decoration-thickness',
+			],
+		},
+	],
+	fix: true,
+
+	reject: [
+		{
+			code: 'a { text-decoration-line: underline; text-decoration-style: solid; text-decoration-color: purple; }',
+			fixed: 'a { text-decoration: underline solid purple; }',
+			description: 'CSS Text Decoration Module Level 3',
+			message: messages.expected('text-decoration'),
+		},
+		{
+			code: 'a { text-decoration-color: purple; text-decoration-thickness: 1px; text-decoration-style: solid; text-decoration-line: underline; text-decoration-width: 1px; }',
+			fixed:
+				'a { text-decoration: underline solid purple; text-decoration-thickness: 1px; text-decoration-width: 1px; }',
```

Comment:
I chose not to add an exception for `text-decoration-width` because #7510 doesn't have a resolution.